cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/JaasConfigTest.java

```diff
@@ -0,0 +1,128 @@
+
+        assertThrows(NullPointerException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testModuleNameContainsEqualSign() {
+        Map<String, String> options = new HashMap<>();
+        options.put("key1", "value1");
+
+        String moduleName = "Module=";
+        assertThrows(IllegalArgumentException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testModuleNameContainsSemicolon() {
+        Map<String, String> options = new HashMap<>();
+        options.put("key1", "value1");
+
+        String moduleName = "Module;";
+        assertThrows(IllegalArgumentException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testKeyContainsEqualSign() {
+        Map<String, String> options = new HashMap<>();
+        options.put("key1=", "value1");
+
+        String moduleName = "Module";
+        assertThrows(IllegalArgumentException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testKeyContainsSemicolon() {
+        Map<String, String> options = new HashMap<>();
+        options.put("key1;", "value1");
+
+        String moduleName = "Module";
+        assertThrows(IllegalArgumentException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testValueContainsEqualSign() {
+        Map<String, String> options = new HashMap<>();
+        options.put("key1", "value=1");
+
+        String moduleName = "Module";
+        assertThrows(IllegalArgumentException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testValueContainsSemicolon() {
+        Map<String, String> options = new HashMap<>();
+        options.put("key1", "value;1");
+
+        String moduleName = "Module";
+        assertThrows(IllegalArgumentException.class, () -> JaasConfig.config(moduleName, options));
+    }
+
+    @Test
+    public void testConfigWithEmptyOptions() {
+        String moduleName = "ExampleModule";
+        Map<String, String> options = new HashMap<>();
+
+        String expectedOutput = "ExampleModule required ;";
+        String result = JaasConfig.config(moduleName, options);
+
+        assertEquals(expectedOutput, result);
+    }
+//    public static void main(String[] a) throws Exception {
+//        //org.apache.kafka.common.security.JaasConfig jaasConfig = new org.apache.kafka.common.security.JaasConfig();
+//        Class<?> aClass = Class.forName("org.apache.kafka.common.security.JaasConfig");
+//
+//        Constructor<?> declaredConstructor = aClass.getDeclaredConstructor(String.class, String.class);
+//        declaredConstructor.setAccessible(true);
+//        Configuration o = (Configuration) declaredConstructor.newInstance("==", "foo required foo=\"value\"with\\\"quote\";");
+////        AppConfigurationEntry[] appConfigurationEntry = o.getAppConfigurationEntry();
+//    }
```

Comment:
Again we usually avoid adding commented code. I think case _I_ understand that we used this to prove/understand what Kafka does and doesn't accept while writing the test. I personally I think there's a case for keeping it, assuming you added a comment explaining what it was for. But I suspect other reviewers might disagree.